### PR TITLE
[utils] ProxySql side-car cleanup in prep for placement

### PR DESCRIPTION
**Pull request by fwiesel**
- A couple of macros were duplicated, they got removed.
- The check for valid modes (unix_socket, host_alias) was fixed
- The version of proxysql gets a patch version update
- Removed the check for Dependency.Enabled, as disabled ones do not
  appear among the dependencies
- The proxysql setting can now point with proxysql.force_enable
  to a list of database settings which are disabled.
  This is needed for proxysql as long as it runs
  with the nova-api database

### CODE REVIEW

**Comment by fwiesel** [comment]
Fixes #3762

File: openstack/utils/templates/_proxysql.tpl [hunk 1]
```diff
@@ -34,7 +34,7 @@
 {{- end }}
 
 {{- define "utils.proxysql.volumes" }}
-  {{- if .Values.proxysql }}
+  {{- if .Values.proxysql -}}
```
Comment (joker-at-work) [inline openstack/utils/templates/_proxysql.tpl:37]: Is this change necessary? I would have assumed the whitespace is consumed by the next `if` anyways.

**Review by joker-at-work — APPROVED** [review]

File: openstack/utils/templates/_proxysql.tpl (hunk 1 above)
Comment (fwiesel) [reply openstack/utils/templates/_proxysql.tpl]: Ah, that one not... I've removed it
